infrastructure/main.tf

```diff
@@ -138,37 +138,60 @@ data "vault_generic_secret" "barApi" {
-  name      = "test-service-secret"
-  value     = "${data.vault_generic_secret.test_s2s_secret.data["value"]}"
-  vault_uri = "${module.key-vault.key_vault_uri}"
-}
-# endregion
+locals {
+  # name of the vault secret holding the key -> name of the corresponding application setting
+  microservice_key_names = {
+    "microservicekey-ccd-admin"                 = "MICROSERVICEKEY_CCD_ADMIN"
+    "microservicekey-ccd-data"                  = "MICROSERVICEKEY_CCD_DATA"
+    "microservicekey-ccd-definition"            = "MICROSERVICEKEY_CCD_DEFINITION"
+    "microservicekey-ccd-gw"                    = "MICROSERVICEKEY_CCD_GW"
+    "microservicekey-ccd-ps"                    = "MICROSERVICEKEY_CCD_PS"
+    "microservicekey-cmc"                       = "MICROSERVICEKEY_CMC"
+    "microservicekey-cmcLegalFrontend"          = "MICROSERVICEKEY_CMC_LEGAL_FRONTEND"
+    "microservicekey-cmcClaimStore"             = "MICROSERVICEKEY_CMC_CLAIM_STORE"
+    "microservicekey-divorce"                   = "MICROSERVICEKEY_DIVORCE"
+    "microservicekey-divorce-frontend"          = "MICROSERVICEKEY_DIVORCE_FRONTEND"
+    "microservicekey-divorceCcdSubmission"      = "MICROSERVICEKEY_DIVORCE_CCD_SUBMISSION"
+    "microservicekey-divorceCcdValidation"      = "MICROSERVICEKEY_DIVORCE_CCD_VALIDATION"
+    "microservicekey-divorceDocumentUpload"     = "MICROSERVICEKEY_DIVORCE_DOCUMENT_UPLOAD"
+    "microservicekey-divorceDocumentGenerator"  = "MICROSERVICEKEY_DIVORCE_DOCUMENT_GENERATOR"
+    "microservicekey-draftStoreTests"           = "MICROSERVICEKEY_DRAFT_STORE_TESTS"
+    "microservicekey-platformJobScheduler"      = "MICROSERVICEKEY_JOBSCHEDULER"
+    "microservicekey-reference"                 = "MICROSERVICEKEY_REFERENCE"
+    "microservicekey-sscs"                      = "MICROSERVICEKEY_SSCS"
+    "microservicekey-sscs-tribunals-case"       = "MICROSERVICEKEY_SSCS_TRIBUNALS_CASE"
+    "microservicekey-probate-frontend"          = "MICROSERVICEKEY_PROBATE_FRONTEND"
+    "microservicekey-probate-backend"           = "MICROSERVICEKEY_PROBATE_BACKEND"
+    "microservicekey-send-letter-consumer"      = "MICROSERVICEKEY_SEND_LETTER_CONSUMER"
+    "microservicekey-send-letter-tests"         = "MICROSERVICEKEY_SEND_LETTER_TESTS"
+    "microservicekey-em-gw"                     = "MICROSERVICEKEY_EM_GW"
+    "microservicekey-finrem"                    = "MICROSERVICEKEY_FINREM"
+    "microservicekey-finrem-draft-store"        = "MICROSERVICEKEY_FINREM_DRAFT_STORE"
+    "microservicekey-jui-webapp"                = "MICROSERVICEKEY_JUI_WEBAPP"
+    "microservicekey-pui-webapp"                = "MICROSERVICEKEY_PUI_WEBAPP"
+    "microservicekey-coh-cor"                   = "MICROSERVICEKEY_COH_COR"
+    "microservicekey-bulk-scan-processor"       = "MICROSERVICEKEY_BULK_SCAN_PROCESSOR"
+    "microservicekey-bulk-scan-processor-tests" = "MICROSERVICEKEY_BULK_SCAN_PROCESSOR_TESTS"
+    "microservicekey-bar-api"                   = "MICROSERVICEKEY_BAR_API"
```

Comment:
could be simplified by storing service name to be used by the app (`MICROSERVICEKEY_` will not be needed and will be added when setting env vars for the app)

Reply:
That's true, I can change that. The secret name is a bit more tricky, though.

Reply:
yeah, I just meant the value is this map, keys would remain the same